DataSources/Oracle/server/aspnet/RevealSdk.Server/Reveal/DataSourceProvider.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Reveal.Sdk;
+using Reveal.Sdk.Data;
+using Reveal.Sdk.Data.Oracle;
+
+namespace RevealSdk.Server.Reveal
+{
+    public class DataSourceProvider : IRVDataSourceProvider
+    {
+        public Task<RVDataSourceItem> ChangeDataSourceItemAsync(IRVUserContext userContext, string dashboardId,
+            RVDataSourceItem dataSourceItem)
+        {
+            if (dataSourceItem is RVOracleDataSourceItem oracleDataSourceItem)
+            {
+                //update underlying data source
+                ChangeDataSourceAsync(userContext, oracleDataSourceItem.DataSource);
+
+                //only change the table if we have selected our custom data source item
+                if (oracleDataSourceItem.Id == "MyOracleDataSourceItem")
+                {
+                    oracleDataSourceItem.Table = "your-table";
+                }
+            }
+
+            return Task.FromResult(dataSourceItem);
+        }
+
+        public Task<RVDashboardDataSource> ChangeDataSourceAsync(IRVUserContext userContext,
+            RVDashboardDataSource dataSource)
+        {
+            //using SID
+            if (dataSource is RVOracleSIDDataSource oracleSidDataSource)
+            {
+                oracleSidDataSource.Host = "your-host";
+                oracleSidDataSource.Database = "your-database";
+                oracleSidDataSource.SID = "your-service-sid";
```

Comment:
What if you just simplified to something like:
```
            if (dataSource is RVOracleSIDDataSource oracleDS)
            {
                oracleDS.Host = "your-host";
                oracleDS.Database = "your-database";
                oracleDS.SID = "your-service-sid";
                //oracleDS.Service = "your-service-name";
            }
```

Basically, just comment out the other appraoch. This way it can simply be uncommented if a customer is using the service instead of the SID